src/EditorFeatures/Core/Shared/Extensions/ITextViewExtensions.cs

```diff
@@ -378,5 +378,12 @@ public static bool TryMoveCaretToAndEnsureVisible(this ITextView textView, Virtu
 
             return span;
         }
+
+        /// <summary>
+        /// Determines if the textbuffer passed in matches the buffer for the textview. If it does
+        /// not, we assume that the textview is in the interactive window.
+        /// </summary>
+        public static bool IsBufferInInteractiveWindow(this ITextView textView, ITextBuffer textBuffer)
+            => textBuffer != textView.TextBuffer;
```

Comment:
this isn't accurate as per the method name.  All this is saying is are we on the root buffer or not?  Other examples that trigger this would be just a normal razor projection view.

Reply:
This is the same hack that was used in inline parameter hints (where I got the code from), I just moved it to be shared so at least we're all being hacky in the same way. The name/comment came from there. Do you know a better way? Better name?

Reply:
see above for better name.

note: this won't work for razor.

Reply:
So the previous comment (which you actually reviewed iirc :P) was wrong? That's fine. I just don't think a name like `IsRootBuffer` is meaningful to intent. Instead it just tells me what the line of code is, and acts like a needless comment. The intent of the code was to check that we weren't performing these actions on the interactive window. Again, is there a better way to check that we're in the interactive window?

Reply:
> Again, is there a better way to check that we're in the interactive window?

I'm not sure.  @tmat may know.

Reply:
so i'm ok with this helper.  but we should just basically say "IsNotSurfaceBufferOfTextView" to at least be honest about what it is doing.

Reply:
`IsNotSurfaceBufferOfTextView` seems fine to me, I'll rename.